packages/p2p/src/pages/buy-sell/currency-dropdown.jsx

```diff
@@ -13,8 +14,21 @@ const CurrencyDropdown = () => {
     const [is_list_visible, setIsListVisible] = React.useState(false);
     const currency_selector_ref = React.useRef(null);
     const { buy_sell_store } = useStores();
-    const { local_currencies, onLocalCurrencySelect, selected_local_currency } = buy_sell_store;
+    const { onLocalCurrencySelect } = buy_sell_store;
     const { showModal } = useModalManagerContext();
+    const {
+        p2p_settings: { currency_list },
+    } = useP2PSettings();
+
+    const local_currencies_list = currency_list.map(currency => ({
+        ...currency,
+        component: (
+            <div className='currency-dropdown__list-item'>
+                <div className='currency-dropdown__list-item-symbol'>{currency.value}</div>
+                <div className='currency-dropdown__list-item-name'>{currency.display_name}</div>
+            </div>
+        ),
+    }));
```

Comment:
had to add the component value here instead of inside the hook as it couldn't recognise it as typeof react element due to the behaviour of the local storage. because we store p2p_settings inside of local_storage, it deserialises the typeof value from the component object, so react won't be able to recognise its a react element.